packages/fuel-indexer-api-server/src/uses.rs

```diff
@@ -208,6 +232,19 @@ pub(crate) async fn register_indexer_assets(
         while let Some(field) = multipart.next_field().await.unwrap() {
             let name = field.name().unwrap_or("").to_string();
             let data = field.bytes().await.unwrap_or_default();
+
+            if params.stop_previous.unwrap_or(false) {
```

Comment:
This should be _after_ I've already successfully uploaded. Else if there is an issue with my upload you will have already removed my already running indexer.

Also, this logic should be outside of the `multipart` loop. Else you will try to run this logic for each multipart field.